src/services/email.service.js

```diff
@@ -0,0 +1,42 @@
+import { config } from 'dotenv';
+import sendgrid from '@sendgrid/mail';
+import Debug from 'debug';
+
+config();
+
+const debug = Debug('dev');
+
+const { SENDGRID_API_KEY, SENDGRID_EMAIL, SENDGRID_ACTIVATION_TEMPLATE } = process.env;
+const templates = { accountActivation: SENDGRID_ACTIVATION_TEMPLATE };
+
+
+/**
+     * Initialises the event .
+     * @param {Object} message - The message payload.
+     * @param {String} type - the type of message which corresponds to the mail template
+     * @param {Object | Array} data - can be a list for bulk sending. must contain email parameter
+     * @return {promise} - Returns a promise that get resolcved to a success object
+*/
+export default async ({ type, data }) => {
+  const template = templates[type];
+
+  if (!template) throw new Error('Email Template not available');
```

Comment:
Is there a more elegant way to handle this error? Or is something else catching it?

Reply:
it is caught when the function is consumed

Reply:
I think will be better to return the error rather than throwing it
Or what do you think?